eradiate/scenes/biosphere.py

```diff
@@ -312,26 +310,29 @@ class HomogeneousDiscreteCanopy(Canopy):
     leaf_positions = attrib_quantity(
         default=[],
         units_compatible=cdu.generator("length"),
+        validator=make_validator_same_length_other("leaf_orientations"),
```

Comment:
I didn't find anything in the attrs docs about comparing two attributes of the same instance. If you know a way to do this without a new validator, please show me :)

Reply:
Or do you mean using the decorator approach that we talked about for the `PerspectiveCameraMeasure`? My thinking was, that this might be a useful validator in other cases as well, that's why I implemented it as a closure.

Reply:
You guessed right :) The way you wrote this is elegant and more maintainable than an ad-hoc component. Actually, now that I think about it, you could have a generic comparison validator accepting a callable for comparison. Still overkill IMO, given that the decorator approach is practicable.